src/libraries/System.Private.Xml/src/System/Xml/Base64Decoder.cs

```diff
@@ -193,44 +180,41 @@ private static byte[] ConstructMapBase64()
                 if (bFilled >= 8)
                 {
                     // get top eight valid bits
-                    *pByte++ = (byte)((b >> (bFilled - 8)) & 0xFF);
+                    bytes[iByte++] = (byte)((b >> (bFilled - 8)) & 0xFF);
                     bFilled -= 8;
 
-                    if (pByte == pBytesEndPos)
+                    if (iByte == bytes.Length)
                     {
                         goto Return;
                     }
                 }
             }
 
-            if (pChar < pCharsEndPos && *pChar == '=')
+            if ((uint)iChar < (uint)chars.Length && chars[iChar] == '=')
```

Comment:
why do you need to cast it everywhere? Should you just make it unsigned from the start?

Reply:
This follows the "bounds check elision" pattern used elsewhere in the runtime code base, where the indexer is cast to uint only for the purposes of comparison against Length. All other uses of the indexer remain as normal signed integers.

I'm not a huge fan of this pattern, since I wish we'd just use `nuint` everywhere and call it a day. :)